from PyQt5 import QtCore, QtGui, QtWidgets
from functools import partial
import sys

class Ui_RadioTelescopeControl(object):
    def __init__(self):
        QtWidgets.QApplication.setStyle(QtWidgets.QStyleFactory.create("Fusion"))

    def setupUi(self, RadioTelescopeControl):
        RadioTelescopeControl.setWindowIcon(QtGui.QIcon('radiotelescope.png'))
        RadioTelescopeControl.setObjectName("RadioTelescopeControl")
        RadioTelescopeControl.resize(429, 416)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        RadioTelescopeControl.setFont(font)
        self.centralwidget = QtWidgets.QWidget(RadioTelescopeControl)
        self.centralwidget.setObjectName("centralwidget")
        self.gridLayout_3 = QtWidgets.QGridLayout(self.centralwidget)
        self.gridLayout_3.setObjectName("gridLayout_3")
        self.tabWidget = QtWidgets.QTabWidget(self.centralwidget)
        self.tabWidget.setStatusTip("")
        self.tabWidget.setObjectName("tabWidget")
        self.tab = QtWidgets.QWidget()
        self.tab.setObjectName("tab")
        self.gridLayout_4 = QtWidgets.QGridLayout(self.tab)
        self.gridLayout_4.setObjectName("gridLayout_4")
        self.groupBox = QtWidgets.QGroupBox(self.tab)
        self.groupBox.setObjectName("groupBox")
        self.gridLayout = QtWidgets.QGridLayout(self.groupBox)
        self.gridLayout.setObjectName("gridLayout")
        self.frame_2 = QtWidgets.QFrame(self.groupBox)
        self.frame_2.setFrameShape(QtWidgets.QFrame.NoFrame)
        self.frame_2.setObjectName("frame_2")
        self.formLayout_2 = QtWidgets.QFormLayout(self.frame_2)
        self.formLayout_2.setObjectName("formLayout_2")
        self.label_4 = QtWidgets.QLabel(self.frame_2)
        self.label_4.setObjectName("label_4")
        self.formLayout_2.setWidget(0, QtWidgets.QFormLayout.LabelRole, self.label_4)
        self.horizontalLayout_2 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_2.setObjectName("horizontalLayout_2")
        self.label_5 = QtWidgets.QLabel(self.frame_2)
        self.label_5.setObjectName("label_5")
        self.horizontalLayout_2.addWidget(self.label_5)
        self.raPosInd = QtWidgets.QLabel(self.frame_2)
        self.raPosInd.setObjectName("raPosInd")
        self.horizontalLayout_2.addWidget(self.raPosInd)
        self.formLayout_2.setLayout(1, QtWidgets.QFormLayout.LabelRole, self.horizontalLayout_2)
        self.horizontalLayout = QtWidgets.QHBoxLayout()
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.label_6 = QtWidgets.QLabel(self.frame_2)
        self.label_6.setObjectName("label_6")
        self.horizontalLayout.addWidget(self.label_6)
        self.decPosInd = QtWidgets.QLabel(self.frame_2)
        self.decPosInd.setObjectName("decPosInd")
        self.horizontalLayout.addWidget(self.decPosInd)
        self.formLayout_2.setLayout(2, QtWidgets.QFormLayout.LabelRole, self.horizontalLayout)
        spacerItem = QtWidgets.QSpacerItem(40, 10, QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        self.formLayout_2.setItem(3, QtWidgets.QFormLayout.SpanningRole, spacerItem)
        self.label_3 = QtWidgets.QLabel(self.frame_2)
        self.label_3.setObjectName("label_3")
        self.formLayout_2.setWidget(4, QtWidgets.QFormLayout.LabelRole, self.label_3)
        self.connectRadioT = QtWidgets.QPushButton(self.frame_2)
        self.connectRadioT.setObjectName("connectRadioT")
        self.formLayout_2.setWidget(5, QtWidgets.QFormLayout.LabelRole, self.connectRadioT)
        self.gridLayout.addWidget(self.frame_2, 0, 0, 1, 1)
        self.gridLayout_4.addWidget(self.groupBox, 0, 0, 1, 1)
        self.groupBox_2 = QtWidgets.QGroupBox(self.tab)
        self.groupBox_2.setObjectName("groupBox_2")
        self.gridLayout_2 = QtWidgets.QGridLayout(self.groupBox_2)
        self.gridLayout_2.setObjectName("gridLayout_2")
        self.stackedWidget = QtWidgets.QStackedWidget(self.groupBox_2)
        self.stackedWidget.setObjectName("stackedWidget")
        self.stackedWidgetPage1 = QtWidgets.QWidget()
        self.stackedWidgetPage1.setObjectName("stackedWidgetPage1")
        self.formLayout = QtWidgets.QFormLayout(self.stackedWidgetPage1)
        self.formLayout.setObjectName("formLayout")
        spacerItem1 = QtWidgets.QSpacerItem(40, 2, QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        self.formLayout.setItem(0, QtWidgets.QFormLayout.LabelRole, spacerItem1)
        self.horizontalLayout_8 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_8.setObjectName("horizontalLayout_8")
        self.stellarConStatText = QtWidgets.QLabel(self.stackedWidgetPage1)
        self.stellarConStatText.setObjectName("stellarConStatText")
        self.horizontalLayout_8.addWidget(self.stellarConStatText)
        self.label_7 = QtWidgets.QLabel(self.stackedWidgetPage1)
        self.label_7.setObjectName("label_7")
        self.horizontalLayout_8.addWidget(self.label_7)
        self.formLayout.setLayout(1, QtWidgets.QFormLayout.LabelRole, self.horizontalLayout_8)
        spacerItem2 = QtWidgets.QSpacerItem(40, 2, QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        self.formLayout.setItem(2, QtWidgets.QFormLayout.LabelRole, spacerItem2)
        self.label_2 = QtWidgets.QLabel(self.stackedWidgetPage1)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.formLayout.setWidget(3, QtWidgets.QFormLayout.LabelRole, self.label_2)
        self.stellariumOperationSelect = QtWidgets.QComboBox(self.stackedWidgetPage1)
        self.stellariumOperationSelect.setStatusTip("")
        self.stellariumOperationSelect.setObjectName("stellariumOperationSelect")
        self.stellariumOperationSelect.addItem("")
        self.stellariumOperationSelect.addItem("")
        self.formLayout.setWidget(4, QtWidgets.QFormLayout.LabelRole, self.stellariumOperationSelect)
        self.label = QtWidgets.QLabel(self.stackedWidgetPage1)
        self.label.setObjectName("label")
        self.formLayout.setWidget(5, QtWidgets.QFormLayout.LabelRole, self.label)
        self.connectStellarium = QtWidgets.QPushButton(self.stackedWidgetPage1)
        self.connectStellarium.setAcceptDrops(False)
        self.connectStellarium.setCheckable(False)
        self.connectStellarium.setObjectName("connectStellarium")
        self.formLayout.setWidget(6, QtWidgets.QFormLayout.LabelRole, self.connectStellarium)
        self.stackedWidget.addWidget(self.stackedWidgetPage1)
        self.stackedWidgetPage2 = QtWidgets.QWidget()
        self.stackedWidgetPage2.setObjectName("stackedWidgetPage2")
        self.formLayout_5 = QtWidgets.QFormLayout(self.stackedWidgetPage2)
        self.formLayout_5.setObjectName("formLayout_5")
        self.label_13 = QtWidgets.QLabel(self.stackedWidgetPage2)
        self.label_13.setObjectName("label_13")
        self.formLayout_5.setWidget(1, QtWidgets.QFormLayout.LabelRole, self.label_13)
        self.horizontalLayout_6 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_6.setObjectName("horizontalLayout_6")
        self.label_15 = QtWidgets.QLabel(self.stackedWidgetPage2)
        self.label_15.setObjectName("label_15")
        self.horizontalLayout_6.addWidget(self.label_15)
        self.raPosInd_2 = QtWidgets.QLabel(self.stackedWidgetPage2)
        self.raPosInd_2.setObjectName("raPosInd_2")
        self.horizontalLayout_6.addWidget(self.raPosInd_2)
        self.formLayout_5.setLayout(2, QtWidgets.QFormLayout.LabelRole, self.horizontalLayout_6)
        self.horizontalLayout_7 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_7.setObjectName("horizontalLayout_7")
        self.label_16 = QtWidgets.QLabel(self.stackedWidgetPage2)
        self.label_16.setObjectName("label_16")
        self.horizontalLayout_7.addWidget(self.label_16)
        self.decPosInd_2 = QtWidgets.QLabel(self.stackedWidgetPage2)
        self.decPosInd_2.setObjectName("decPosInd_2")
        self.horizontalLayout_7.addWidget(self.decPosInd_2)
        self.formLayout_5.setLayout(3, QtWidgets.QFormLayout.LabelRole, self.horizontalLayout_7)
        spacerItem3 = QtWidgets.QSpacerItem(40, 5, QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        self.formLayout_5.setItem(0, QtWidgets.QFormLayout.SpanningRole, spacerItem3)
        self.stackedWidget.addWidget(self.stackedWidgetPage2)
        self.gridLayout_2.addWidget(self.stackedWidget, 0, 0, 1, 1)
        self.gridLayout_4.addWidget(self.groupBox_2, 0, 1, 1, 1)
        self.groupBox_3 = QtWidgets.QGroupBox(self.tab)
        self.groupBox_3.setCheckable(False)
        self.groupBox_3.setObjectName("groupBox_3")
        self.formLayout_3 = QtWidgets.QFormLayout(self.groupBox_3)
        self.formLayout_3.setObjectName("formLayout_3")
        self.horizontalLayout_4 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_4.setObjectName("horizontalLayout_4")
        self.label_12 = QtWidgets.QLabel(self.groupBox_3)
        self.label_12.setObjectName("label_12")
        self.horizontalLayout_4.addWidget(self.label_12)
        self.latTextInd = QtWidgets.QLabel(self.groupBox_3)
        self.latTextInd.setObjectName("latTextInd")
        self.horizontalLayout_4.addWidget(self.latTextInd)
        self.formLayout_3.setLayout(0, QtWidgets.QFormLayout.LabelRole, self.horizontalLayout_4)
        self.horizontalLayout_3 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_3.setObjectName("horizontalLayout_3")
        self.label_10 = QtWidgets.QLabel(self.groupBox_3)
        self.label_10.setObjectName("label_10")
        self.horizontalLayout_3.addWidget(self.label_10)
        self.lonTextInd = QtWidgets.QLabel(self.groupBox_3)
        self.lonTextInd.setObjectName("lonTextInd")
        self.horizontalLayout_3.addWidget(self.lonTextInd)
        self.formLayout_3.setLayout(1, QtWidgets.QFormLayout.LabelRole, self.horizontalLayout_3)
        self.horizontalLayout_5 = QtWidgets.QHBoxLayout()
        self.horizontalLayout_5.setObjectName("horizontalLayout_5")
        self.label_14 = QtWidgets.QLabel(self.groupBox_3)
        self.label_14.setObjectName("label_14")
        self.horizontalLayout_5.addWidget(self.label_14)
        self.altTextInd = QtWidgets.QLabel(self.groupBox_3)
        self.altTextInd.setObjectName("altTextInd")
        self.horizontalLayout_5.addWidget(self.altTextInd)
        self.formLayout_3.setLayout(2, QtWidgets.QFormLayout.LabelRole, self.horizontalLayout_5)
        self.locatChange = QtWidgets.QPushButton(self.groupBox_3)
        self.locatChange.setObjectName("locatChange")
        self.formLayout_3.setWidget(4, QtWidgets.QFormLayout.LabelRole, self.locatChange)
        spacerItem4 = QtWidgets.QSpacerItem(40, 10, QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        self.formLayout_3.setItem(3, QtWidgets.QFormLayout.SpanningRole, spacerItem4)
        self.gridLayout_4.addWidget(self.groupBox_3, 1, 0, 1, 1)
        self.groupBox_4 = QtWidgets.QGroupBox(self.tab)
        self.groupBox_4.setObjectName("groupBox_4")
        self.formLayout_4 = QtWidgets.QFormLayout(self.groupBox_4)
        self.formLayout_4.setObjectName("formLayout_4")
        self.label_8 = QtWidgets.QLabel(self.groupBox_4)
        self.label_8.setObjectName("label_8")
        self.formLayout_4.setWidget(1, QtWidgets.QFormLayout.LabelRole, self.label_8)
        self.pushButton = QtWidgets.QPushButton(self.groupBox_4)
        self.pushButton.setObjectName("pushButton")
        self.formLayout_4.setWidget(6, QtWidgets.QFormLayout.LabelRole, self.pushButton)
        self.onTargetProgress = QtWidgets.QProgressBar(self.groupBox_4)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.onTargetProgress.sizePolicy().hasHeightForWidth())
        self.onTargetProgress.setSizePolicy(sizePolicy)
        self.onTargetProgress.setProperty("value", 10)
        self.onTargetProgress.setInvertedAppearance(False)
        self.onTargetProgress.setObjectName("onTargetProgress")
        self.formLayout_4.setWidget(4, QtWidgets.QFormLayout.SpanningRole, self.onTargetProgress)
        self.label_11 = QtWidgets.QLabel(self.groupBox_4)
        self.label_11.setObjectName("label_11")
        self.formLayout_4.setWidget(2, QtWidgets.QFormLayout.LabelRole, self.label_11)
        self.movTextInd = QtWidgets.QLabel(self.groupBox_4)
        self.movTextInd.setObjectName("movTextInd")
        self.formLayout_4.setWidget(2, QtWidgets.QFormLayout.FieldRole, self.movTextInd)
        spacerItem5 = QtWidgets.QSpacerItem(40, 10, QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        self.formLayout_4.setItem(5, QtWidgets.QFormLayout.SpanningRole, spacerItem5)
        self.trackTextInd = QtWidgets.QLabel(self.groupBox_4)
        self.trackTextInd.setObjectName("trackTextInd")
        self.formLayout_4.setWidget(1, QtWidgets.QFormLayout.FieldRole, self.trackTextInd)
        self.gridLayout_4.addWidget(self.groupBox_4, 1, 1, 1, 1)
        self.tabWidget.addTab(self.tab, "")
        self.tab_2 = QtWidgets.QWidget()
        self.tab_2.setObjectName("tab_2")
        self.tabWidget.addTab(self.tab_2, "")
        self.gridLayout_3.addWidget(self.tabWidget, 0, 0, 2, 1)
        RadioTelescopeControl.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(RadioTelescopeControl)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 355, 21))
        self.menubar.setObjectName("menubar")
        self.menuThis_is_a_menu = QtWidgets.QMenu(self.menubar)
        self.menuThis_is_a_menu.setObjectName("menuThis_is_a_menu")
        self.menuTools = QtWidgets.QMenu(self.menubar)
        self.menuTools.setObjectName("menuTools")
        self.menuAbout = QtWidgets.QMenu(self.menubar)
        self.menuAbout.setObjectName("menuAbout")
        self.menuTCP = QtWidgets.QMenu(self.menubar)
        self.menuTCP.setObjectName("menuTCP")
        RadioTelescopeControl.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(RadioTelescopeControl)
        self.statusbar.setObjectName("statusbar")
        RadioTelescopeControl.setStatusBar(self.statusbar)
        self.actionOpen = QtWidgets.QAction(RadioTelescopeControl)
        self.actionOpen.setObjectName("actionOpen")
        self.actionSettings = QtWidgets.QAction(RadioTelescopeControl)
        self.actionSettings.setObjectName("actionSettings")
        self.actionLicense = QtWidgets.QAction(RadioTelescopeControl)
        self.actionLicense.setObjectName("actionLicense")
        self.actionAbout = QtWidgets.QAction(RadioTelescopeControl)
        self.actionAbout.setObjectName("actionAbout")
        self.actionExit = QtWidgets.QAction(RadioTelescopeControl)
        self.actionExit.setObjectName("actionExit")
        self.actionExit.triggered.connect(partial(self.close_application, object = RadioTelescopeControl))
        self.actionSave = QtWidgets.QAction(RadioTelescopeControl)
        self.actionSave.setObjectName("actionSave")
        self.actionCalibration = QtWidgets.QAction(RadioTelescopeControl)
        self.actionCalibration.setObjectName("actionCalibration")
        self.actionLocation = QtWidgets.QAction(RadioTelescopeControl)
        self.actionLocation.setShortcutVisibleInContextMenu(True)
        self.actionLocation.setObjectName("actionLocation")
        self.actionManual_Control = QtWidgets.QAction(RadioTelescopeControl)
        self.actionManual_Control.setObjectName("actionManual_Control")
        self.menuThis_is_a_menu.addAction(self.actionOpen)
        self.menuThis_is_a_menu.addAction(self.actionSave)
        self.menuThis_is_a_menu.addSeparator()
        self.menuThis_is_a_menu.addAction(self.actionExit)
        self.menuTools.addAction(self.actionCalibration)
        self.menuTools.addAction(self.actionLocation)
        self.menuTools.addAction(self.actionManual_Control)
        self.menuAbout.addAction(self.actionAbout)
        self.menuAbout.addAction(self.actionLicense)
        self.menuTCP.addAction(self.actionSettings)
        self.menubar.addAction(self.menuThis_is_a_menu.menuAction())
        self.menubar.addAction(self.menuTCP.menuAction())
        self.menubar.addAction(self.menuTools.menuAction())
        self.menubar.addAction(self.menuAbout.menuAction())

        self.retranslateUi(RadioTelescopeControl)
        self.tabWidget.setCurrentIndex(0)
        self.stackedWidget.setCurrentIndex(0)
        QtCore.QMetaObject.connectSlotsByName(RadioTelescopeControl)

    def retranslateUi(self, RadioTelescopeControl):
        _translate = QtCore.QCoreApplication.translate
        RadioTelescopeControl.setWindowTitle(_translate("RadioTelescopeControl", "Radio Telescope Control"))
        self.groupBox.setTitle(_translate("RadioTelescopeControl", "Radio Telescope"))
        self.label_4.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600; text-decoration: underline;\">Dish Position</span></p></body></html>"))
        self.label_5.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600;\">RA:</span></p></body></html>"))
        self.raPosInd.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" color:#ff0000;\">0h 0m 0s</span></p></body></html>"))
        self.label_6.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600;\">DEC:</span></p></body></html>"))
        self.decPosInd.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" color:#ff0000;\">0</span><span style=\" color:#ff0000; vertical-align:super;\">o</span><span style=\" color:#ff0000;\"> 0\' 0\'\'</span></p></body></html>"))
        self.label_3.setText(_translate("RadioTelescopeControl", "TCP Connection"))
        self.connectRadioT.setText(_translate("RadioTelescopeControl", "Connect"))
        self.groupBox_2.setTitle(_translate("RadioTelescopeControl", "Stellarium"))
        self.stellarConStatText.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600;\">Status:</span></p></body></html>"))
        self.label_7.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" color:#ff0000;\">Disconnected</span></p></body></html>"))
        self.label_2.setText(_translate("RadioTelescopeControl", "Slew Command"))
        self.stellariumOperationSelect.setItemText(0, _translate("RadioTelescopeControl", "Transit"))
        self.stellariumOperationSelect.setItemText(1, _translate("RadioTelescopeControl", "Aim and track"))
        self.label.setText(_translate("RadioTelescopeControl", "TCP Server"))
        self.connectStellarium.setText(_translate("RadioTelescopeControl", "Enable"))
        self.label_13.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600; text-decoration: underline;\">Selected Object</span></p></body></html>"))
        self.label_15.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600;\">RA:</span></p></body></html>"))
        self.raPosInd_2.setText(_translate("RadioTelescopeControl", "0h 0m 0s"))
        self.label_16.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600;\">DEC:</span></p></body></html>"))
        self.decPosInd_2.setText(_translate("RadioTelescopeControl", "<html><head/><body><p>0<span style=\" vertical-align:super;\">o</span> 0\' 0\'\'</p></body></html>"))
        self.groupBox_3.setTitle(_translate("RadioTelescopeControl", "Location"))
        self.label_12.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600;\">Lat:</span></p></body></html>"))
        self.latTextInd.setText(_translate("RadioTelescopeControl", "<html><head/><body><p align=\"center\">0<span style=\" vertical-align:super;\">o</span> 0\' 0.00\'\'</p></body></html>"))
        self.label_10.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600;\">Lon:</span></p></body></html>"))
        self.lonTextInd.setText(_translate("RadioTelescopeControl", "<html><head/><body><p align=\"center\">0<span style=\" vertical-align:super;\">o</span> 0\' 0.00\'\'</p></body></html>"))
        self.label_14.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600;\">Alt:</span></p></body></html>"))
        self.altTextInd.setText(_translate("RadioTelescopeControl", "<html><head/><body><p align=\"center\">0m</p></body></html>"))
        self.locatChange.setText(_translate("RadioTelescopeControl", "Change"))
        self.groupBox_4.setTitle(_translate("RadioTelescopeControl", "State"))
        self.label_8.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600;\">Tracking:</span></p></body></html>"))
        self.pushButton.setText(_translate("RadioTelescopeControl", "Stop"))
        self.label_11.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" font-weight:600;\">Moving:</span></p></body></html>"))
        self.movTextInd.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" color:#ff0000;\">No</span></p></body></html>"))
        self.trackTextInd.setText(_translate("RadioTelescopeControl", "<html><head/><body><p><span style=\" color:#ff0000;\">No</span></p></body></html>"))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.tab), _translate("RadioTelescopeControl", "Connection"))
        self.tabWidget.setTabToolTip(self.tabWidget.indexOf(self.tab), _translate("RadioTelescopeControl", "Connection control"))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.tab_2), _translate("RadioTelescopeControl", "Receiver"))
        self.tabWidget.setTabToolTip(self.tabWidget.indexOf(self.tab_2), _translate("RadioTelescopeControl", "Receiver control"))
        self.menuThis_is_a_menu.setTitle(_translate("RadioTelescopeControl", "File"))
        self.menuTools.setTitle(_translate("RadioTelescopeControl", "Tools"))
        self.menuAbout.setTitle(_translate("RadioTelescopeControl", "About"))
        self.menuTCP.setTitle(_translate("RadioTelescopeControl", "TCP"))
        self.actionOpen.setText(_translate("RadioTelescopeControl", "Open"))
        self.actionOpen.setStatusTip(_translate("RadioTelescopeControl", "Open a settings file"))
        self.actionOpen.setShortcut(_translate("RadioTelescopeControl", "Ctrl+O"))
        self.actionSettings.setText(_translate("RadioTelescopeControl", "Settings"))
        self.actionSettings.setStatusTip(_translate("RadioTelescopeControl", "Change the TCP settings"))
        self.actionSettings.setShortcut(_translate("RadioTelescopeControl", "Ctrl+T"))
        self.actionLicense.setText(_translate("RadioTelescopeControl", "License"))
        self.actionLicense.setStatusTip(_translate("RadioTelescopeControl", "View the application licensing"))
        self.actionAbout.setText(_translate("RadioTelescopeControl", "About..."))
        self.actionAbout.setStatusTip(_translate("RadioTelescopeControl", "View application description"))
        self.actionExit.setText(_translate("RadioTelescopeControl", "Exit"))
        self.actionExit.setStatusTip(_translate("RadioTelescopeControl", "Exit the program"))
        self.actionExit.setShortcut(_translate("RadioTelescopeControl", "Ctrl+E"))
        self.actionSave.setText(_translate("RadioTelescopeControl", "Save Settings"))
        self.actionSave.setStatusTip(_translate("RadioTelescopeControl", "Save the current program settings"))
        self.actionSave.setShortcut(_translate("RadioTelescopeControl", "Ctrl+S"))
        self.actionCalibration.setText(_translate("RadioTelescopeControl", "Calibration"))
        self.actionCalibration.setStatusTip(_translate("RadioTelescopeControl", "Calibrate the radio telescope\'s dish positioning"))
        self.actionLocation.setText(_translate("RadioTelescopeControl", "Location"))
        self.actionLocation.setStatusTip(_translate("RadioTelescopeControl", "Change the location and location settings"))
        self.actionLocation.setShortcut(_translate("RadioTelescopeControl", "Ctrl+L"))
        self.actionManual_Control.setText(_translate("RadioTelescopeControl", "Manual Control"))
        self.actionManual_Control.setStatusTip(_translate("RadioTelescopeControl", "Manually control the dish\'s position"))
        self.actionManual_Control.setShortcut(_translate("RadioTelescopeControl", "Ctrl+M"))
        
    def close_application(self, object):
        choice = QtWidgets.QMessageBox.question(object, 'Exit', "Are you sure?", QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No)
        if choice == QtWidgets.QMessageBox.Yes:
            sys.exit()
        else:
            pass
        
class Ui_TCPSettings(object):
    def setupUi(self, TCPSettings):
        TCPSettings.setWindowIcon(QtGui.QIcon('Net.png'))
        TCPSettings.setObjectName("TCPSettings")
        TCPSettings.resize(240, 320)
        self.centralwidget = QtWidgets.QWidget(TCPSettings)
        self.centralwidget.setObjectName("centralwidget")
        TCPSettings.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(TCPSettings)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 240, 21))
        self.menubar.setObjectName("menubar")
        TCPSettings.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(TCPSettings)
        self.statusbar.setObjectName("statusbar")
        TCPSettings.setStatusBar(self.statusbar)
        
        extractAction = QtWidgets.QAction(QtGui.QIcon('TCP.png'), 'Python', TCPSettings)
        
        self.toolBar = QtWidgets.QToolBar(TCPSettings)
        self.toolBar.setEnabled(True)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Preferred, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.toolBar.sizePolicy().hasHeightForWidth())
        self.toolBar.setSizePolicy(sizePolicy)
        self.toolBar.setToolButtonStyle(QtCore.Qt.ToolButtonIconOnly)
        self.toolBar.setObjectName("toolBar")
        self.toolBar.addAction(extractAction) #Create a toolbar
        TCPSettings.addToolBar(QtCore.Qt.TopToolBarArea, self.toolBar)

        self.retranslateUi(TCPSettings)
        QtCore.QMetaObject.connectSlotsByName(TCPSettings)

    def retranslateUi(self, TCPSettings):
        _translate = QtCore.QCoreApplication.translate
        TCPSettings.setWindowTitle(_translate("TCPSettings", "TCP Settings"))
        self.toolBar.setWindowTitle(_translate("TCPSettings", "toolBar"))
